server/index.js

```diff
@@ -130,16 +180,21 @@ app.post('/api/add/waitlist', (req, res, next) => {
       values ($1,$2,(select "userId" from "newUser"))
       returning *
       `;
-      const params = [addcustname, addcustmobile, password, 'Customer'];
-      db.query(sql, params)
-        .then(result => {
-          const [customerAdded] = result.rows;
-          if (!customerAdded) {
-            throw new ClientError(401, 'mobile number is already used');
-          }
-          res.json(customerAdded);
+          const params = [addcustname, addcustmobile, hashedPassword, 'Customer'];
+          db.query(sql, params)
+            .then(result => {
+              const [customerAdded] = result.rows;
+              if (!customerAdded) {
+                throw new ClientError(401, 'mobile number is already used');
+              }
+              res.json(customerAdded);
+            })
+            .catch(err => next(err));
         })
-        .catch(err => next(err));
+        .catch(err => {
+          console.error(err);
+        });
```

Comment:
Be sure to always send the error to `next()` so that the client does not hang and timeout.